dali/python/nvidia/dali/external_source.py

```diff
@@ -464,6 +475,20 @@ def __call__(
                 raise ValueError(
                     "``prefetch_queue_depth`` must be a positive integer, got {}.".format(
                         prefetch_queue_depth))
+            if source_desc.kind == _SourceKind.CALLABLE:
+                if not source_desc.has_inputs:
```

Comment:
What if it accepts more than one argument?

Reply:
Than it's an error either way, but via the External Source - it only accepts 0 and 1-argument callables.

Reply:
Understood.